tests/integration/test_charm.py

```diff
@@ -60,6 +60,9 @@ async def test_relate_dependencies(ops_test: OpsTest):
         config={"kind": "ingress"},
         trust=True,
     )
+    await ops_test.model.add_relation(
+        "istio-pilot:istio-pilot", "istio-ingressgateway:istio-pilot"
+    )
 
```

Comment:
You might want to add this to make sure istio won't get in the way of your tests:

```python
    await ops_test.model.wait_for_idle(
        ["istio-pilot", "istio-ingressgateway"],
        raise_on_blocked=False,
        status="active",
        timeout=90 * 10,
    )
```